.github/workflows/docker-publish-site.yaml

```
@@ -0,0 +1,65 @@
+name: Build and Push Site
+
+on:
+  push:
+    branches: 
+      - "master"
+    paths:
+      - "containers/dockerfiles/site.Dockerfile"
```

I think you should watch for changes in the install scripts as well, so that when something changes (e.g. add/remove packages) it will trigger the workflow.
```suggestion
      - "containers/dockerfiles/site.Dockerfile"
      - "containers/scripts/install_site.sh"
```